doc/user/tutorials/making_your_first_app.rst

```diff
@@ -0,0 +1,510 @@
+
+    (armi) $ python -c "import armi;armi.configure()"
+
+You should see an ARMI splash-screen and an ARMI version print out. If you do, you are ready
+to proceed.
+
+.. tip:: If you are having trouble getting it installed, see :ref:`getting-help`. You may
+    need to ensure your ``PYTHON_PATH`` variable includes the armi installation directory.
+
+.. note:: This tutorial is a companion for the :doc:`/developer/making_armi_based_apps`
+    developer documentation.
+
+Starting a new app
+==================
+ARMI-based applications can take on many forms, depending on your workflow. Examples may include:
+
+* Application and plugins together under one folder
+* Application in one folder, plugins in separate ones
+
+For simplicity, we will build an application that contains one plugin that runs
+neutronics and thermal hydraulics in one folder. This architecture will be a good starting
+point for many projects, and can always be separated if needed.
+
+From the command line, ``cd`` into a new directory where you'd like to store your
+application code. Make a folder structure that works as a `normal Python package
+<https://packaging.python.org/tutorials/packaging-projects/>`_, and create some empty
+files for us to fill in, like this::
+
+    my_package/
+        myapp/
+            __init__.py
+            __main__.py
+            app.py
+            plugin.py
+            fluxSolver.py
+            thermalSolver.py
+        setup.py
+        README.md
+        LICENSE.md
+
+
+These files are:
+
+* The outer :file:`my_package` root directory is a container for your app. The name does not
+  matter to ARMI; you can rename it to anything.
+
+* The inner :file:`myapp` directory is the actual Python package for your app. Its name is the
+  Python package name you will use to import anything inside (e.g. ``myapp.plugins``).
+
+* :file:`myapp/__init__.py` tells Python that this directory is a Python package and registers
+  the application with the ARMI framework.
+
+* :file:`myapp/__main__.py` provides one or more entry points for users of your app (including
+  you!) to start running it.
+
+* :file:`myapp/app.py` contains the actual app registration code that will be called by
+  :file:`__init__.py`. This can be named anything as long as it is consistent with the
+  registration code in :file:`myapp/__init__.py`. 
+
+* :file:`myapp/plugin.py` contains the code that defines the physics plugin we will create
+
+* :file:`myapp/fluxSolver.py` contains the flux solver
+
+* :file:`myapp/thermalSolver.py` contains the thermal/hydraulics solver
+
+* :file:`setup.py` the python package installation file to help users install your
+  application.
+
+* :file:`README.md` and :file:`LICENSE.md` are an optional description and license of your
+  application that would be prominently featured, e.g. in a github repo, if you were to
+  put it there.
+
+Registering the app with ARMI
+=============================
```

Comment:
As I am reading through this the first time, it seems like registering the app within ARMI should be placed after definining the `app` class. Mainly because the `ExampleApp` doesn't exist yet.

Reply:
I kind of oscillated between writing code first and connecting it later and back a few times.